### PR TITLE
Destroy Subscribes when finished

### PR DESCRIPTION
**Pull request by surya-prakash-singh1**
Destroy Subscribes when finished

### CODE REVIEW

File: feature-libs/pickup-in-store/root/components/container/pickup-in-store-order-consignment/pickup-in-store-order-consignment-container.component.ts [hunk 1]
```diff
@@ -4,33 +4,42 @@
...
  */
 
-import { Component, OnInit, Optional } from '@angular/core';
+import { Component, OnDestroy, OnInit, Optional } from '@angular/core';
 import { Consignment, Order } from '@spartacus/order/root';
 import { OutletContextData } from '@spartacus/storefront';
+import { Subscription } from 'rxjs';
 import { tap } from 'rxjs/operators';
+
+export type IOutletContextData = { item: Consignment; order: Order };
```
Comment (WeizhengSap) [inline feature-libs/pickup-in-store/root/components/container/pickup-in-store-order-consignment/pickup-in-store-order-consignment-container.component.ts:13]: Can you check  `IOutletContextData` in develop branch? I remembered `order` was removed.

File: feature-libs/pickup-in-store/root/components/container/pickup-in-store-order-consignment/pickup-in-store-order-consignment-container.component.ts [hunk 2]
```diff
@@ -4,33 +4,42 @@
...
+
 /**
  * A container component of the pair of the pickup options radio buttons for cart entry.
  */
 @Component({
   selector: 'cx-pickup-in-store-order-consignment',
   templateUrl: './pickup-in-store-order-consignment-container.component.html',
 })
-export class PickupInStoreOrderConsignmentContainerComponent implements OnInit {
+export class PickupInStoreOrderConsignmentContainerComponent
```
Comment (WeizhengSap) [inline feature-libs/pickup-in-store/root/components/container/pickup-in-store-order-consignment/pickup-in-store-order-consignment-container.component.ts:22]: You need update this class. This class in develop branch is modified a lot.

Comment (surya-prakash-singh1) [reply feature-libs/pickup-in-store/root/components/container/pickup-in-store-order-consignment/pickup-in-store-order-consignment-container.component.ts:22]: Hi Wei, these changed are part of https://github.com/SAP/spartacus/pull/17176 and will be added once it's merged to release branch as well. They are merged as part of the different changes to develop hence we are replicating the same as asked.

File: feature-libs/pickup-in-store/root/components/container/pickup-in-store-order-consignment/pickup-in-store-order-consignment-container.component.ts (hunk 1 above)
Comment (surya-prakash-singh1) [reply feature-libs/pickup-in-store/root/components/container/pickup-in-store-order-consignment/pickup-in-store-order-consignment-container.component.ts:13]: yes right, these changed are part of https://github.com/SAP/spartacus/pull/17176 and will be added once it's merged to the release branch as well. They are merged as part of the different changes to develop hence we are replicating the same as asked.

**Review by WeizhengSap — APPROVED** [review]

**Pull request merged**